### ISSUE TITLE
rust 1.77.0 build failure

### ISSUE THREAD

**Issue by chenrui333** [MitMaro/git-interactive-rebase-tool#906]
👋 when [upgrading rust to 1.77.0](https://github.com/Homebrew/homebrew-core/pull/166865), we ran into some regression build failure, ie the lint `unused_tuple_struct_fields` has been renamed to `dead_code`

```
error: lint `unused_tuple_struct_fields` has been renamed to `dead_code`
  --> src/runtime/src/lib.rs:51:2
   |
51 |     unused_tuple_struct_fields,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use the new name: `dead_code`
   |
note: the lint level is defined here
  --> src/runtime/src/lib.rs:14:2
   |
14 |     warnings
   |     ^^^^^^^^
   = note: `#[deny(renamed_and_removed_lints)]` implied by `#[deny(warnings)]`

error: could not compile `girt-runtime` (lib) due to 1 previous error
```

**Comment by MitMaro** [comment]
Thanks for the post!

I've recently converted all linting errors to warnings by default, so hopefully this will be less of a problem in the future for projects like Homebrew.

I will also address the particular error, to use the new name.

For Homebrew, I will need to put out a new release, since I believe it's pinned to the last release tag. I cannot make any promises on when a release will be cut, but given there is an impact on Homebrew I will attempt to find some time.

**Comment by jubalh** [comment]
Would you mind creating a new release with the fixes for this issue?

**Comment by MitMaro** [comment]
Hey @jubalh , sadly I ran out of time at the end of March, but I've started working on cutting a new release with the progress being tracking in #915.

Hopefully the testing will go smoothly, and a new release will be available soon.

**Comment by MitMaro** [comment]
I've just released version 2.4.0, which should address the build issues in homebrew.

As part of this release, I've greatly loosened the lint checks that occur during a regular build, which means that Rust updates should generally work without requiring a new release. Going forward, these errors should instead be warnings, and better adhere to the Rust semver guarantees.

Apologies for causing problems with the update in Homebrew package. I'm a long-term user of homebrew and greatly appreciate all the great work from the project maintainers for keeping packages up to date, and want to reduce that effort as much as possible.

**Comment by chenrui333** [comment]
thanks @MitMaro 

close via https://github.com/Homebrew/homebrew-core/pull/174519
